front/lib/multi_regions/region_lookup_client.ts

```diff
@@ -0,0 +1,78 @@
+import { EnvironmentConfig, isDevelopment } from "@dust-tt/types";
+
+import type {
+  UserLookupRequestBodyType,
+  UserLookupResponse,
+} from "@app/pages/api/lookup/[resource]";
+
+const CLUSTER_REGIONS = ["local", "europe-west1", "us-central1"] as const;
+type ClusterRegionType = (typeof CLUSTER_REGIONS)[number];
+
+const REGIONS_TO_URLS: Map<ClusterRegionType, string> = new Map([
```

Comment:
`nit`: We usually use Record in this case

Reply:
I got somehow the opposite feedback from @flvndvd in the past as `Record` lose some types informations, as well as do not raise a warning if you lookup a value that is undefined (`record[keythatdonotexists].something` will not raise warning).

The only downside of `Map` (afaik) is that it's not serializable.

Is there any reason to use Record over Map ?

Reply:
I'm not that deep into TS tbh, my remark was mainly for consistency because it is the first time I come across the use of a Map in such case. But not able to judge whether one or the other is better suited here 😇